### DISCUSSION TITLE
Help with Calling OnlyOwner Function in Foundry and Forge 0.2.0 for Smart Contract Testing

### DISCUSSION THREAD

**Discussion by Giulianomlodi** [Cyfrin/foundry-full-course-cu#1132] [Q&A] [answered]
Hello everyone,

I'm currently working on testing my smart contract using Foundry and Forge version 0.2.0. I've encountered a specific issue related to calling a function with the OnlyOwner modifier. My main challenge is determining how to programmatically find the deployer (owner) address during unit tests.

Here's what I've done so far:

I've set up my testing environment using Foundry and Forge 0.2.0.
I'm attempting to call a function in my smart contract that is restricted by the OnlyOwner modifier.
I need to identify the deployer's address within my test scripts to simulate the owner's interaction with the contract.
I've conducted thorough searches on Google and consulted AI assistants, but I haven't found a solution or guidance specific to my scenario.

Could anyone guide me on how to programmatically find the deployer's address in this context, or provide an example of how to call a function with the OnlyOwner modifier during unit testing?

Any insights or resources that could assist me in resolving this issue would be greatly appreciated.

Thank you for your time and help!

Best regards,

**Comment by 0xusmanf** [comment, marked as answer]
If you are using Ownable contract from openzeppelin-contracts then you can use `owner()` function to get the address of the owner and then prank call the desired function with it.

Here is an example:
#### ExampleContract.sol
```solidity
import { Ownable } from "openzeppelin/contracts/access/Ownable.sol";

contract ExampleContract is Ownable {

    uint256 public counter;

    constructor() Ownable(msg.sender) {
    }

    function addOne() external onlyOwner {
        counter = counter + 1;
    }
}
```
#### ExampleContractTest.t.sol
```solidity
import { Test } from "forge-std/Test.sol";
import { ExampleContract } from "../src/ExampleContract.sol";

contract ExampleContractTest is Test {

    ExampleContract exampleContract;

    function setUp() public {
        exampleContract = new ExampleContract();
    }

    function testIfOwnerCanCallAddOne() public {
        vm.prank(exampleContract.owner());
        exampleContract.addOne();
        assertEq(exampleContract.counter,1);
    }
}
```

**Reply by Giulianomlodi** [reply]
Thanks so much! ♥️

**Reply by 0xusmanf** [reply]
Happy to help mate!
